xarray/tests/test_combine.py

```diff
@@ -366,8 +366,9 @@ def test_nested_concat(self):
         assert_identical(expected, actual)
 
         objs = [Dataset({"x": [0], "y": [0]}), Dataset({"x": [0]})]
-        with pytest.raises(KeyError):
-            combine_nested(objs, concat_dim="x")
+        actual = combine_nested(objs, concat_dim="x")
+        expected = Dataset({"x": [0, 0], "y": [0]})
```

Comment:
seems sensible to me.

Reply:
In general if you concatenate two different numbers then you're also verifying the order was retained, but it's not really important.

Reply:
good point. Fixed.